lib/active_merchant/billing/gateways/nmi.rb

```diff
@@ -146,6 +146,8 @@ def add_payment_method(post, payment_method, options)
           post[:token_cryptogram] = payment_method.payment_cryptogram
         elsif(card_brand(payment_method) == 'check')
           post[:payment] = 'check'
+          post[:firstname] = payment_method.first_name
+          post[:lastname] = payment_method.last_name
```

Comment:
I wasn't sure at first either, but both `CreditCard` and `Check` have first and lastname:
https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/credit_card.rb#L117-L122
https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/check.rb#L9